container/lxd/client.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package lxd
+
+import (
+	"github.com/juju/errors"
+	"github.com/lxc/lxd/client"
+)
+
+// Client extends the upstream LXD client.
+type Client struct {
+	ImageServer
+}
+
+// NewClient builds and returns a Client for high-level interaction with the
+// input LXD container server.
+func NewClient(svr lxd.ContainerServer) *Client {
+	return &Client{
+		ImageServer: ImageServer{svr},
+	}
+}
+
+// UpdateServerConfig updates the server configuration with the input values.
+func (c *Client) UpdateServerConfig(cfg map[string]interface{}) error {
+	svr, eTag, err := c.GetServer()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if svr.Config == nil {
+		svr.Config = cfg
+	} else {
+		for k, v := range cfg {
+			svr.Config[k] = v
+		}
+	}
+	return errors.Trace(c.UpdateServer(svr.Writable(), eTag))
+}
+
+// UpdateContainerConfig updates the configuration for the container with the
+// input name, using the input values.
+func (c *Client) UpdateContainerConfig(name string, cfg map[string]string) error {
+	container, eTag, err := c.GetContainer(name)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if container.Config == nil {
```

Comment:
Similarly here.

Reply:
Interesting that this config map is string to string, but above it is string to interface. Why the inconsistency?

Reply:
From a scan of the LXD code, config values can be int/bool/string, but they are validated based on incoming strings. Not sure why it is different between server and container...